### DISCUSSION TITLE
Centrifugal (from the inside out) blur of a composed frame?

### DISCUSSION THREAD

**Discussion by decent-numar** [ImageMagick/ImageMagick#7219] [Digital Image Processing]
Hello, I need to process images that sometimes contain unwanted artifacts close to the edges. With images with mostly white background, I'm simply adding a **white frame**:

`magick.exe "image.jpg" -shave 30x30 -bordercolor white -border 30 "image.jpg"`

With images that do not contain white borders, I'm adding a **blurred frame**:

```
$border = 30 #thickness of the frame
magick.exe convert "image.jpg" `( +clone -blur 100x100 -fill white -colorize 20% `) `( +clone -gamma 0 -shave $borderx$border -bordercolor white -border $borderx$border `) -composite `( +clone -gamma 0 -shave $borderx$border -bordercolor black -border $borderx$border `) -compose screen -composite "image.jpg"
```

Such blur preserves the background color, but also has two downsides:

1. The artifacts are still there, albeit blurred.
2. The frame itself is not blurred. The edge between an image and its frame is often visible.

I would like to compose such a frame that would take pixels from its inner edge and spread them in the outward direction. That would achieve my goals: preserve the background color and get rid of the artifacts behind the frame.

Then I'd like to know how to blur the inner edge of the frame itself.

I hope this makes sense. Thanks.

**Comment by snibgo** [comment]
> ...  unwanted artifacts close to the edges.
The artifacts are still there, albeit blurred.

Please show sample inputs that have artifacts.

**Reply by decent-numar** [reply]
This is an image with artifacts (black stips on the edges). Ideally, I want to get rid of them as much as possible.
![color background](https://github.com/ImageMagick/ImageMagick/assets/166003622/63645483-633d-4611-81c8-f49dfab3183d)

This image has a blurred frame (command in the OP above). It is somewhat better for my purposes, but I'd prefer if the frame was blurred and if the darker spots that actually are blurred artifacts disappeared.
![color background blur](https://github.com/ImageMagick/ImageMagick/assets/166003622/0672ff01-8cb6-47d1-85c2-aeebfdc14e7d)

So I imagined this solution... spread pixels from the inner part of the frame out.

**Comment by snibgo** [comment]
I suggest you first remove the artifacts, such as by this command, where `centrif.jpg` is your input file:
```
magick centrif.jpg -write mpr:INP ( +clone -channel Red -separate -threshold 80% +write x.png ) -alpha off -compose CopyOpacity -composite -virtual-pixel None -blur 0x5 x2.png
```
